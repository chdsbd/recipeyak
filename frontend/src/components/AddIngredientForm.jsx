import React from 'react'

import { ButtonPrimary } from './Buttons'

const AddIngredientForm = ({
  handleAddIngredient,
  cancelAddIngredient,
  handleInputChange,
  quantity,
  name,
  description,
  optional,
  loading,
  error,
}) =>
  <form onSubmit={ async e => {
    e.preventDefault()
    await handleAddIngredient()
    document.querySelector('#firstinput').focus()
  }
  }>
  <div className="add-ingredient-grid mb-2 mt-2">

    <div>
      <input
        id="firstinput"
        onChange={ handleInputChange }
        onFocus={ e => e.target.select() }
        autoFocus
        value={ quantity }
        className={ 'my-input' + (error ? ' is-danger' : '') }
        type="text"
        placeholder="3 lbs"
        name="quantity"/>
    </div>

    <div>
      <input
        onChange={ handleInputChange }
        onFocus={ e => e.target.select() }
        value={ name }
        className={ 'my-input' + (error ? ' is-danger' : '') }
        type="text"
        placeholder="tomato"
        name="name"/>
    </div>

    <div className="grid-entire-row">
      <input
        onChange={ handleInputChange }
        onFocus={ e => e.target.select() }
        value={ description }
        className={ 'my-input' + (error ? ' is-danger' : '') }
        type="text"
        placeholder="diced at 3cm"
        name="description"/>
      { error
          ? <p className="fs-4 c-danger">A recipe needs at least one ingredient</p>
          : null
      }
    </div>
  </div>

  <label className="d-flex align-items-center cursor-pointer mb-2">
  <input
    onChange={ handleInputChange }
    checked={ optional }
    name="optional"
    type='checkbox'
    className="mr-2"/>
    Optional
  </label>

  <div className="field is-grouped">
    <p className="control">
      <ButtonPrimary
        disabled={ quantity === '' && name === '' }
        className="is-small"
        type="submit"
        name="add ingredient"
        loading={ loading }>
        Add
      </ButtonPrimary>
    </p>
    <p className="control">
      <input
        onClick={ cancelAddIngredient }
        className="my-button is-small"
        type="button"
        name="cancel add ingredient"
        value="cancel"/>
    </p>
  </div>
</form>

export default AddIngredientForm
