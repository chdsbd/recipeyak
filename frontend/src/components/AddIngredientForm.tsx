import * as React from 'react'

import { ButtonPrimary } from './Buttons'

const AddIngredientForm = ({
  handleAddIngredient,
  cancelAddIngredient,
  handleInputChange,
  quantity,
  name,
  description,
  loading,
  error
}) =>
  <form onSubmit={ async e => {
    e.preventDefault()
    await handleAddIngredient()
    document.querySelector('#firstinput').focus()
  }
  }>
  <div className="add-ingredient-grid mb-2 mt-2">

    <div>
      <input
        id="firstinput"
        onChange={ handleInputChange }
        onFocus={ e => e.target.select() }
        value={ quantity }
        className={ 'my-input' + (error ? ' is-danger' : '') }
        type="text"
        placeholder="3 lbs"
        name="quantity"/>
    </div>

    <div>
      <input
        onChange={ handleInputChange }
        onFocus={ e => e.target.select() }
        value={ name }
        className={ 'my-input' + (error ? ' is-danger' : '') }
        type="text"
        placeholder="tomato"
        name="name"/>
    </div>

    <div className="grid-entire-row">
      <input
        onChange={ handleInputChange }
        onFocus={ e => e.target.select() }
        value={ description }
        className={ 'my-input' + (error ? ' is-danger' : '') }
        type="text"
        placeholder="diced at 3cm"
        name="description"/>
      { error
          ? <p className="fs-4 c-danger">A recipe needs at least one ingredient</p>
          : null
      }
    </div>
  </div>

  <div className="field is-grouped">
    <p className="control">
      <ButtonPrimary
        disabled={ quantity === '' && name === '' }
        type="submit"
        name="add ingredient"
        loading={ loading }>
        Add
      </ButtonPrimary>
    </p>
    <p className="control">
      { quantity !== '' || name !== ''
        ? <input
            disabled={ quantity === '' && name === '' }
            onClick={ cancelAddIngredient }
            className="my-button"
            type="button"
            name="cancel add ingredient"
            value="âœ•"/>
        : null
      }
    </p>
  </div>
</form>

export default AddIngredientForm
