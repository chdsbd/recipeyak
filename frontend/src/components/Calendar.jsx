import React from "react";
import { connect } from "react-redux";
import startOfMonth from "date-fns/start_of_month";
import endOfMonth from "date-fns/end_of_month";
import eachDay from "date-fns/each_day";
import addMonths from "date-fns/add_months";
import subMonths from "date-fns/sub_months";
import subDays from "date-fns/sub_days";
import addDays from "date-fns/add_days";
import format from "date-fns/format";
import PropTypes from "prop-types";

import {
  fetchCalendar,
  fetchTeams,
  fetchShoppingList,
  fetchRecipeList,
  setScheduleURL
} from "../store/actions";

import { pyFormat, daysFromSunday, daysUntilSaturday } from "../date";

import { teamsFrom } from "../store/mapState";

import { push } from "react-router-redux";

import { ButtonPrimary, ButtonPlain } from "./Buttons";
import Loader from "./Loader";
import CalendarDay from "./CalendarDay";

function monthYearFromDate(date) {
  return format(date, "MMM | YYYY");
}

const mapStateToProps = (state, props) => {
  const isTeam = props.match.params.id != null;
  const teamID = isTeam ? parseInt(props.match.params.id, 10) : "personal";

  const days = state.calendar.allIds
    .map(id => state.calendar[id])
    .filter(x => {
      if (!isTeam) {
        // we know that if there is a userID, it will be the current user's
        return x.user != null;
      }
      return x.team === teamID;
    })
    .reduce(
      (a, b) => ({
        ...a,
        [b.on]: {
          ...a[b.on],
          [b.id]: b
        }
      }),
      {}
    );

  return {
    days,
    loading: state.calendar.loading,
    error: state.calendar.error,
    loadingTeams: state.teams.loading,
    teams: teamsFrom(state),
    teamID,
    isTeam,
    start: state.shoppinglist.startDay,
    end: state.shoppinglist.endDay
  };
};

const mapDispatchToProps = dispatch => ({
  fetchData: (month, teamID = "personal") =>
    dispatch(fetchCalendar(teamID, month)),
  fetchTeams: () => dispatch(fetchTeams()),
  navTo: url => {
    dispatch(push(url));
    dispatch(setScheduleURL(url));
  },
  refetchShoppingListAndRecipes: (teamID, start, end) => {
    return Promise.all([
      dispatch(fetchRecipeList(teamID)),
      dispatch(fetchShoppingList(teamID, start, end))
    ]);
  }
});

@connect(
  mapStateToProps,
  mapDispatchToProps
)
export default class Calendar extends React.Component {
  state = {
    month: new Date(),
    initialLoad: false,
    owner: "personal"
  };

  static propTypes = {
    loadingTeams: PropTypes.bool.isRequired,
    loading: PropTypes.bool.isRequired,
    error: PropTypes.bool.isRequired,
    className: PropTypes.string.isRequired,
    fetchData: PropTypes.func.isRequired,
    fetchTeams: PropTypes.func.isRequired,
    navTo: PropTypes.func.isRequired,
    teams: PropTypes.arrayOf(PropTypes.object).isRequired,
    isTeam: PropTypes.bool.isRequired,
    teamID: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
      .isRequired,
    refetchShoppingListAndRecipes: PropTypes.func.isRequired
  };

  static defaultProps = {
    loading: true,
    error: false,
    className: "",
    loadingTeams: true,
    teams: [],
    teamID: "personal"
  };

  componentDidMount() {
    this.props.fetchTeams();
    this.props
      .fetchData(this.state.month, this.props.teamID)
      .then(() => this.setState({ initialLoad: true }));
  }

  refetchData = (teamID = this.props.teamID) => {
    this.props.fetchData(this.state.month, teamID);
    this.props.refetchShoppingListAndRecipes(
      teamID,
      this.props.startDay,
      this.props.endDay
    );
  };

  prevMonth = () => {
    this.setState(({ month }) => ({
      month: subMonths(month, 1)
    }));
    this.props.fetchData(this.state.month, this.props.teamID);
  };

  nextMonth = () => {
    this.setState(({ month }) => ({
      month: addMonths(month, 1)
    }));
    this.props.fetchData(this.state.month, this.props.teamID);
  };

  currentMonth = () => {
    this.setState({ month: new Date() });
  };

  handleOwnerChange = e => {
    const teamID = e.target.value;
    const url = teamID === "personal" ? "/schedule/" : `/t/${teamID}/schedule/`;

    const isRecipes = this.props.match.params["type"] === "recipes";

    const ending = isRecipes ? "recipes" : "shopping";

    const urlWithEnding = url + ending;

    // navTo is async so we can't count on the URL to have changed by the time we refetch the data
    this.props.navTo(urlWithEnding);
    this.refetchData(teamID);
  };

  render() {
    if (this.props.error) {
      return <p>Error fetching data</p>;
    }

    if (this.props.loading && !this.state.initialLoad) {
      return (
        <div
          className={`d-flex w-100 justify-content-center align-items-center ${
            this.props.className
          }`}
        >
          <div>
            <Loader />
          </div>
        </div>
      );
    }

    return (
      <div className={`flex-grow-1 ${this.props.className}`}>
        <div className="d-flex justify-space-between align-items-end">
          <div className="d-flex align-items-center">
            <p title={this.state.month.toString()}>
              {monthYearFromDate(this.state.month)}
            </p>
            <div className="select is-small ml-2">
              <select
                onChange={this.handleOwnerChange}
                value={this.props.teamID}
                disabled={this.props.loadingTeams}
              >
                <option value="personal">Personal</option>
                {this.props.teams.map(t => (
                  <option key={t.id} value={t.id}>
                    {t.name}
                  </option>
                ))}
              </select>
            </div>
          </div>

          <div>
            <ButtonPlain className="is-small" onClick={this.prevMonth}>
              {"←"}
            </ButtonPlain>
            <ButtonPrimary
              className="ml-1 mr-1 is-small"
              onClick={this.currentMonth}
            >
              Today
            </ButtonPrimary>
            <ButtonPlain className="is-small" onClick={this.nextMonth}>
              {"→"}
            </ButtonPlain>
          </div>
        </div>
        <div
          className={
            "d-grid grid-gap-1 calendar-grid grid-auto-rows-unset mb-0"
          }
        >
          <b>Su</b>
          <b>Mo</b>
          <b>Tu</b>
          <b>We</b>
          <b>Th</b>
          <b>Fr</b>
          <b>Sa</b>
        </div>
        <div className={"d-grid grid-gap-1 calendar-grid mb-0"}>
          {eachDay(
            subDays(
              startOfMonth(this.state.month),
              daysFromSunday(this.state.month)
            ),
            addDays(
              endOfMonth(this.state.month),
              daysUntilSaturday(endOfMonth(this.state.month))
            )
          ).map(date => (
            <CalendarDay
              item={this.props.days[pyFormat(date)]}
              date={date}
              key={date}
              teamID={this.props.teamID}
            />
          ))}
        </div>
        <p className="mt-1">
          press <kbd>?</kbd> for help
        </p>
      </div>
    );
  }
}
