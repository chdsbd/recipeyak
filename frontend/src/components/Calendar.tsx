import React from "react"
import { connect } from "react-redux"
import startOfMonth from "date-fns/start_of_month"
import endOfMonth from "date-fns/end_of_month"
import eachDay from "date-fns/each_day"
import addMonths from "date-fns/add_months"
import subMonths from "date-fns/sub_months"
import subDays from "date-fns/sub_days"
import addDays from "date-fns/add_days"
import format from "date-fns/format"

import {
  fetchCalendar,
  fetchTeams,
  fetchShoppingList,
  fetchRecipeList,
  Dispatch
} from "../store/actions"

import { pyFormat, daysFromSunday, daysUntilSaturday } from "../date"

import { teamsFrom } from "../store/mapState"

import { push } from "react-router-redux"

import { ButtonPrimary, ButtonPlain } from "./Buttons"
import Loader from "./Loader"
import CalendarDay from "./CalendarDay"
import { RootState } from "../store/store"
import { ITeam } from "../store/reducers/teams"
import { ICalRecipe } from "../store/reducers/calendar"
import { ScheduleRouteParams } from "./Schedule"

function monthYearFromDate(date: Date) {
  return format(date, "MMM | YYYY")
}

interface IDays {
  readonly [key: string]: ICalRecipe
}

const mapStateToProps = (state: RootState, props: ICalendarProps) => {
  const isTeam = props.teamID !== "personal"

  const days = state.calendar.allIds
    // TODO(chdsbd): Remove string | number ids
    .map((id: number | string) => state.calendar[id as number])
    .filter((x: ICalRecipe) => {
      if (!isTeam) {
        // we know that if there is a userID, it will be the current user's
        return x.user != null
      }
      return x.team === props.teamID
    })
    .reduce(
      (a, b) => ({
        ...a,
        [b.on]: {
          ...a[b.on],
          [b.id]: b
        }
      }),
      {} as IDays
    )

  return {
    days,
    loading: state.calendar.loading,
    error: state.calendar.error,
    loadingTeams: !!state.teams.loading,
    teams: teamsFrom(state),
    teamID: props.teamID,
    isTeam,
    start: state.shoppinglist.startDay,
    end: state.shoppinglist.endDay
  }
}

const mapDispatchToProps = (dispatch: Dispatch) => ({
  fetchData: fetchCalendar(dispatch),
  fetchTeams: fetchTeams(dispatch),
  navTo: (url: string) => {
    dispatch(push(url))
  },
  refetchShoppingListAndRecipes: (
    teamID: ITeam["id"] | "personal",
    start: Date,
    end: Date
  ) => {
    return Promise.all([
      fetchRecipeList(dispatch)(teamID),
      fetchShoppingList(dispatch)(teamID, start, end)
    ])
  }
})

interface ICalendarProps extends ScheduleRouteParams {
  readonly loadingTeams: boolean
  readonly loading: boolean
  readonly error: boolean
  readonly className: string
  readonly fetchTeams: () => void
  readonly navTo: (url: string) => void
  readonly fetchData: (
    teamID: ITeam["id"] | "personal",
    month: Date
  ) => Promise<void>
  readonly refetchShoppingListAndRecipes: (
    teamID: ITeam["id"] | "personal",
    startDay: Date,
    endDay: Date
  ) => void
  readonly teams: ReadonlyArray<ITeam>
  readonly days: IDays
  readonly isTeam: boolean
  readonly teamID: ITeam["id"] | "personal"
  readonly startDay: Date
  readonly endDay: Date
  readonly type: "shopping" | "recipes"
}

interface ICalendarState {
  readonly month: Date
  readonly initialLoad: boolean
  readonly owner: ITeam["id"] | "personal"
}

class Calendar extends React.Component<ICalendarProps, ICalendarState> {
  readonly state: ICalendarState = {
    month: new Date(),
    initialLoad: false,
    owner: "personal"
  }

  componentDidMount() {
    this.props.fetchTeams()
    this.props
      .fetchData(this.props.teamID, this.state.month)
      .then(() => this.setState({ initialLoad: true }))
  }

  readonly refetchData = (
    teamID: ITeam["id"] | "personal" = this.props.teamID
  ) => {
    this.props.fetchData(teamID, this.state.month)
    this.props.refetchShoppingListAndRecipes(
      teamID,
      this.props.startDay,
      this.props.endDay
    )
  }

  readonly prevMonth = () => {
    this.setState(({ month }) => ({
      month: subMonths(month, 1)
    }))
    this.props.fetchData(this.props.teamID, this.state.month)
  }

  readonly nextMonth = () => {
    this.setState(({ month }) => ({
      month: addMonths(month, 1)
    }))
    this.props.fetchData(this.props.teamID, this.state.month)
  }

  readonly currentMonth = () => {
    this.setState({ month: new Date() })
  }

  readonly handleOwnerChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
    const teamID =
      e.target.value === "personal" ? "personal" : parseInt(e.target.value, 10)
    const url = teamID === "personal" ? "/schedule/" : `/t/${teamID}/schedule/`

    const isRecipes = this.props.type === "recipes"

    const ending = isRecipes ? "recipes" : "shopping"

    const urlWithEnding = url + ending

    // navTo is async so we can't count on the URL to have changed by the time we refetch the data
    this.props.navTo(urlWithEnding)
    this.refetchData(teamID)
  }

  render() {
    if (this.props.error) {
      return <p>Error fetching data</p>
    }

    if (this.props.loading && !this.state.initialLoad) {
      return (
        <div
          className={`d-flex w-100 justify-content-center align-items-center ${
            this.props.className
          }`}>
          <div>
            <Loader />
          </div>
        </div>
      )
    }

    return (
      <div className={`flex-grow-1 ${this.props.className}`}>
        <div className="d-flex justify-space-between align-items-end">
          <div className="d-flex align-items-center">
            <p title={this.state.month.toString()}>
              {monthYearFromDate(this.state.month)}
            </p>
            <div className="select is-small ml-2">
              <select
                onChange={this.handleOwnerChange}
                value={this.props.teamID}
                disabled={this.props.loadingTeams}>
                <option value="personal">Personal</option>
                {this.props.teams.map(t => (
                  <option key={t.id} value={t.id}>
                    {t.name}
                  </option>
                ))}
              </select>
            </div>
          </div>

          <div>
            <ButtonPlain className="is-small" onClick={this.prevMonth}>
              {"←"}
            </ButtonPlain>
            <ButtonPrimary
              className="ml-1 mr-1 is-small"
              onClick={this.currentMonth}>
              Today
            </ButtonPrimary>
            <ButtonPlain className="is-small" onClick={this.nextMonth}>
              {"→"}
            </ButtonPlain>
          </div>
        </div>
        <div
          className={
            "d-grid grid-gap-1 calendar-grid grid-auto-rows-unset mb-0"
          }>
          <b>Su</b>
          <b>Mo</b>
          <b>Tu</b>
          <b>We</b>
          <b>Th</b>
          <b>Fr</b>
          <b>Sa</b>
        </div>
        <div className={"d-grid grid-gap-1 calendar-grid mb-0"}>
          {eachDay(
            subDays(
              startOfMonth(this.state.month),
              daysFromSunday(this.state.month)
            ),
            addDays(
              endOfMonth(this.state.month),
              daysUntilSaturday(endOfMonth(this.state.month))
            )
          ).map(date => (
            <CalendarDay
              item={this.props.days[pyFormat(date)]}
              date={date}
              key={date.toString()}
              teamID={this.props.teamID}
            />
          ))}
        </div>
        <p className="mt-1">
          press <kbd>?</kbd> for help
        </p>
      </div>
    )
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Calendar)
