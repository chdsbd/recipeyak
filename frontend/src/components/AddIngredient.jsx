import React from 'react'
import AddIngredientForm from './AddIngredientForm'

class AddIngredient extends React.Component {
  constructor (props) {
    super(props)

    this.emptyState = {
      addingIngredient: false,
      quantity: '',
      unit: '',
      name: '',
      description: ''
    }

    this.state = this.emptyState
  }

  handleInputChange = e => {
    this.setState({ [e.target.name]: e.target.value })
  }

  handleAddIngredient = e => {
    e.preventDefault()
    this.addIngredient(this.props.id, this.state.ingredient)
  }

  addingIngredient = () => {
    this.setState({ addingIngredient: true })
  }

  cancelAddIngredient = () => {
    this.setState(this.emptyState)
  }

  clearInputs = () => {
    this.setState({ quantity: '', unit: '', name: '', description: '' })
  }

  addIngredient = async (id, ingredient) => {
    // TODO: pass this prop in from Recipe or via store
    const { quantity, unit, name, description } = this.state
    await this.props.addIngredient(id, { quantity, unit, name, description })
    this.clearInputs()
  }

  render () {
    const { state } = this

    const units = [
      'large',
      'small',
      'medium',
      'tsp',
      'tbsp',
      'cup',
      'lb',
      'kg',
      'mg',
      'gram',
      'liter',
      'ml',
      'gal.',
      'oz',
      'fl oz',
      'pint',
      'quart',
      'pinch',
      'inch',
      'cm',
      'm',
      'mm'
    ]

    return (
      state.addingIngredient
      ? <AddIngredientForm
          handleAddIngredient={this.handleAddIngredient}
          cancelAddIngredient={this.cancelAddIngredient}
          handleInputChange={this.handleInputChange}
          units={units}
          quantity={this.state.quantity}
          unit={this.state.unit}
          name={this.state.name}
          description={this.state.description}
        />
      : <p className="flex-center">
        <button
          onClick={ this.addingIngredient }
          className="button is-link">
          Add another
        </button>
      </p>
    )
  }
}

export default AddIngredient
