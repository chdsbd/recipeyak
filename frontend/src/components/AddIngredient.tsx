import React from "react"
import PropTypes from "prop-types"
import AddIngredientForm from "./AddIngredientForm"

export interface IIngredientBasic {
  quantity: string
  name: string
  description: string
}

interface IAddIngredientProps {
  readonly onCancel: () => void
  readonly id: number
  readonly addIngredient: (
    id: number,
    { quantity, name, description }: IIngredientBasic
  ) => void
  readonly loading: boolean
  readonly autoFocus: boolean
}

interface IAddIngredientState {
  readonly quantity: string
  readonly name: string
  readonly description: string
  readonly optional: boolean
  readonly addingIngredient: boolean
}

export type AddIngredientFields = Exclude<
  keyof IAddIngredientState,
  "addingIngredient"
>

export default class AddIngredient extends React.Component<
  IAddIngredientProps,
  IAddIngredientState
> {
  static propTypes = {
    id: PropTypes.number.isRequired,
    addIngredient: PropTypes.func.isRequired,
    loading: PropTypes.bool.isRequired,
    autoFocus: PropTypes.bool,
    onCancel: PropTypes.func.isRequired
  }

  static defaultProps = {
    loading: false
  }

  emptyState: IAddIngredientState = {
    quantity: "",
    name: "",
    description: "",
    optional: false,
    addingIngredient: false
  }

  state: IAddIngredientState = this.emptyState

  handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    this.setState(({
      [e.target.name]: e.target.value
    } as unknown) as IAddIngredientState)
  }

  addingIngredient = () => {
    this.setState({ addingIngredient: true })
  }

  cancelAddIngredient = () => {
    this.props.onCancel()
    this.setState(this.emptyState)
  }

  clearInputs = () => {
    this.setState(this.emptyState)
  }

  render() {
    const { clearInputs, cancelAddIngredient, handleInputChange } = this
    const { id, addIngredient, loading, autoFocus } = this.props
    const { quantity, name, description, optional } = this.state

    return (
      <AddIngredientForm
        handleAddIngredient={async () => {
          await addIngredient(id, { quantity, name, description })
          clearInputs()
        }}
        loading={loading}
        cancelAddIngredient={cancelAddIngredient}
        handleInputChange={handleInputChange}
        quantity={quantity}
        name={name}
        description={description}
        optional={optional}
        autoFocus={autoFocus}
      />
    )
  }
}
