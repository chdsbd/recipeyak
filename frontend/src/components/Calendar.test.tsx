import React from "react"
import renderer from "react-test-renderer"
import { TestProvider, mockDate } from "@/testUtils"
import { Calendar } from "@/components/Calendar"

describe("<Calendar> Snap", () => {
  beforeAll(() => {
    mockDate.set("1776-1-1")
  })
  afterAll(() => {
    mockDate.reset()
  })
  test("smoke test render with empty data", () => {
    const tree = renderer
      .create(
        <TestProvider>
          <Calendar
            loadingTeams={false}
            error={false}
            fetchTeams={jest.fn()}
            navTo={jest.fn()}
            fetchData={jest.fn()}
            teams={[]}
            days={{}}
            teamID={10}
            type="recipes"
            refetchShoppingListAndRecipes={jest.fn()}
          />
        </TestProvider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
