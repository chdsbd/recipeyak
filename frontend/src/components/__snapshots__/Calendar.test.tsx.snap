// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar> Snap smoke test render with empty data 1`] = `
.c3 {
  -webkit-flex: 1 1 0%;
  -ms-flex: 1 1 0%;
  flex: 1 1 0%;
  padding: 0.25rem;
  background-color: whitesmoke;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

.c3:not(:last-child) {
  margin-right: 0.25rem;
}

.c4 {
  -webkit-flex: 1 1 0%;
  -ms-flex: 1 1 0%;
  flex: 1 1 0%;
  padding: 0.25rem;
  background-color: whitesmoke;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
  border-bottom: 2px solid #ff7247;
}

.c4:not(:last-child) {
  margin-right: 0.25rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c1 > b {
  width: 14.285714285714285%;
}

.c1 > b:not(:last-child) {
  margin-right: 0.25rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 33.33333333333333%;
}

.c2:not(:last-child) {
  margin-bottom: 0.25rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

@media (max-width:992px) {
  .c3:not(:last-child) {
    margin-right: 0;
    margin-bottom: 0.25rem;
  }
}

@media (max-width:992px) {
  .c3 {
    width: 100%;
    min-height: 75px;
  }
}

@media (max-width:992px) {
  .c4:not(:last-child) {
    margin-right: 0;
    margin-bottom: 0.25rem;
  }
}

@media (max-width:992px) {
  .c4 {
    width: 100%;
    min-height: 75px;
  }
}

@media (max-width:992px) {
  .c1 {
    display: none;
  }
}

@media (max-width:992px) {
  .c2 {
    height: 100%;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    margin-top: 0.5rem;
  }

  .c2:first-child,
  .c2:last-child {
    display: none;
  }
}

@media (max-width:630px) {
  .c0 {
    height: 100%;
  }
}

<div
  className="c0"
>
  <p
    className="mr-2"
    title="Sun Dec 31 1775 00:00:00 GMT+0000 (Coordinated Universal Time)"
  >
    Dec 31 | 1775
  </p>
  <section
    className="d-flex flex-grow justify-space-between"
  >
    <div
      className="select is-small  undefined"
    >
      <select
        className=""
        disabled={false}
        multiple={undefined}
        onChange={[Function]}
        value={10}
      >
        <option
          value="personal"
        >
          Personal
        </option>
      </select>
    </div>
    <section>
      <button
        className="my-button is-small"
        onClick={[Function]}
      >
        ←
      </button>
      <button
        className="my-button ml-1 mr-1 is-small"
        onClick={[Function]}
      >
        Today
      </button>
      <button
        className="my-button is-small"
        onClick={[Function]}
      >
        →
      </button>
    </section>
  </section>
  <div
    className="c1"
  >
    <b>
      Sun
    </b>
    <b>
      Mon
    </b>
    <b>
      Tue
    </b>
    <b>
      Wed
    </b>
    <b>
      Thu
    </b>
    <b>
      Fri
    </b>
    <b>
      Sat
    </b>
  </div>
  <section
    className="mb-2 flex-grow-1 h-100"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <div
          className="d-flex"
        >
          <div
            className="d-none d-medium-block"
          >
            <span>
              Sun
            </span>
            <span
              className="mx-1"
            >
              ∙
            </span>
          </div>
          <span>
            24
          </span>
        </div>
        <ul />
      </div>
      <div
        className="c3"
      >
        <div
          className="d-flex"
        >
          <div
            className="d-none d-medium-block"
          >
            <span>
              Mon
            </span>
            <span
              className="mx-1"
            >
              ∙
            </span>
          </div>
          <span>
            25
          </span>
        </div>
        <ul />
      </div>
      <div
        className="c3"
      >
        <div
          className="d-flex"
        >
          <div
            className="d-none d-medium-block"
          >
            <span>
              Tue
            </span>
            <span
              className="mx-1"
            >
              ∙
            </span>
          </div>
          <span>
            26
          </span>
        </div>
        <ul />
      </div>
      <div
        className="c3"
      >
        <div
          className="d-flex"
        >
          <div
            className="d-none d-medium-block"
          >
            <span>
              Wed
            </span>
            <span
              className="mx-1"
            >
              ∙
            </span>
          </div>
          <span>
            27
          </span>
        </div>
        <ul />
      </div>
      <div
        className="c3"
      >
        <div
          className="d-flex"
        >
          <div
            className="d-none d-medium-block"
          >
            <span>
              Thu
            </span>
            <span
              className="mx-1"
            >
              ∙
            </span>
          </div>
          <span>
            28
          </span>
        </div>
        <ul />
      </div>
      <div
        className="c3"
      >
        <div
          className="d-flex"
        >
          <div
            className="d-none d-medium-block"
          >
            <span>
              Fri
            </span>
            <span
              className="mx-1"
            >
              ∙
            </span>
          </div>
          <span>
            29
          </span>
        </div>
        <ul />
      </div>
      <div
        className="c3"
      >
        <div
          className="d-flex"
        >
          <div
            className="d-none d-medium-block"
          >
            <span>
              Sat
            </span>
            <span
              className="mx-1"
            >
              ∙
            </span>
          </div>
          <span>
            30
          </span>
        </div>
        <ul />
      </div>
    </div>
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <div
          className="d-flex"
        >
          <div
            className="d-none d-medium-block"
          >
            <span>
              Sun
            </span>
            <span
              className="mx-1"
            >
              ∙
            </span>
          </div>
          <span>
            31
          </span>
        </div>
        <ul />
      </div>
      <div
        className="c4"
      >
        <div
          className="d-flex"
        >
          <div
            className="d-none d-medium-block"
          >
            <span>
              Mon
            </span>
            <span
              className="mx-1"
            >
              ∙
            </span>
          </div>
          <span>
            Jan 1
          </span>
        </div>
        <ul />
      </div>
      <div
        className="c3"
      >
        <div
          className="d-flex"
        >
          <div
            className="d-none d-medium-block"
          >
            <span>
              Tue
            </span>
            <span
              className="mx-1"
            >
              ∙
            </span>
          </div>
          <span>
            2
          </span>
        </div>
        <ul />
      </div>
      <div
        className="c3"
      >
        <div
          className="d-flex"
        >
          <div
            className="d-none d-medium-block"
          >
            <span>
              Wed
            </span>
            <span
              className="mx-1"
            >
              ∙
            </span>
          </div>
          <span>
            3
          </span>
        </div>
        <ul />
      </div>
      <div
        className="c3"
      >
        <div
          className="d-flex"
        >
          <div
            className="d-none d-medium-block"
          >
            <span>
              Thu
            </span>
            <span
              className="mx-1"
            >
              ∙
            </span>
          </div>
          <span>
            4
          </span>
        </div>
        <ul />
      </div>
      <div
        className="c3"
      >
        <div
          className="d-flex"
        >
          <div
            className="d-none d-medium-block"
          >
            <span>
              Fri
            </span>
            <span
              className="mx-1"
            >
              ∙
            </span>
          </div>
          <span>
            5
          </span>
        </div>
        <ul />
      </div>
      <div
        className="c3"
      >
        <div
          className="d-flex"
        >
          <div
            className="d-none d-medium-block"
          >
            <span>
              Sat
            </span>
            <span
              className="mx-1"
            >
              ∙
            </span>
          </div>
          <span>
            6
          </span>
        </div>
        <ul />
      </div>
    </div>
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <div
          className="d-flex"
        >
          <div
            className="d-none d-medium-block"
          >
            <span>
              Sun
            </span>
            <span
              className="mx-1"
            >
              ∙
            </span>
          </div>
          <span>
            7
          </span>
        </div>
        <ul />
      </div>
      <div
        className="c3"
      >
        <div
          className="d-flex"
        >
          <div
            className="d-none d-medium-block"
          >
            <span>
              Mon
            </span>
            <span
              className="mx-1"
            >
              ∙
            </span>
          </div>
          <span>
            8
          </span>
        </div>
        <ul />
      </div>
      <div
        className="c3"
      >
        <div
          className="d-flex"
        >
          <div
            className="d-none d-medium-block"
          >
            <span>
              Tue
            </span>
            <span
              className="mx-1"
            >
              ∙
            </span>
          </div>
          <span>
            9
          </span>
        </div>
        <ul />
      </div>
      <div
        className="c3"
      >
        <div
          className="d-flex"
        >
          <div
            className="d-none d-medium-block"
          >
            <span>
              Wed
            </span>
            <span
              className="mx-1"
            >
              ∙
            </span>
          </div>
          <span>
            10
          </span>
        </div>
        <ul />
      </div>
      <div
        className="c3"
      >
        <div
          className="d-flex"
        >
          <div
            className="d-none d-medium-block"
          >
            <span>
              Thu
            </span>
            <span
              className="mx-1"
            >
              ∙
            </span>
          </div>
          <span>
            11
          </span>
        </div>
        <ul />
      </div>
      <div
        className="c3"
      >
        <div
          className="d-flex"
        >
          <div
            className="d-none d-medium-block"
          >
            <span>
              Fri
            </span>
            <span
              className="mx-1"
            >
              ∙
            </span>
          </div>
          <span>
            12
          </span>
        </div>
        <ul />
      </div>
      <div
        className="c3"
      >
        <div
          className="d-flex"
        >
          <div
            className="d-none d-medium-block"
          >
            <span>
              Sat
            </span>
            <span
              className="mx-1"
            >
              ∙
            </span>
          </div>
          <span>
            13
          </span>
        </div>
        <ul />
      </div>
    </div>
  </section>
  <p
    className="mt-2 mb-1 hide-sm"
  >
    press 
    <kbd>
      ?
    </kbd>
     for help
  </p>
</div>
`;
